packages/i18n-utils/test/localize-url.js

```diff
@@ -0,0 +1,333 @@
+		[
+			'https://wordpress.com/',
+			'https://de.wordpress.com/',
+			'https://wordpress.com/start/',
+			'https://wordpress.com/wp-login.php?action=lostpassword',
+		].forEach( ( fullUrl ) => {
+			getLocaleSlug.mockImplementationOnce( () => 'en' );
+			expect( localizeUrl( fullUrl ) ).toEqual( fullUrl );
+		} );
+	} );
+
+	test( 'should not change relative URLs', () => {
+		[ '/me/account', '/settings' ].forEach( ( fullUrl ) => {
+			getLocaleSlug.mockImplementationOnce( () => 'en' );
+			expect( localizeUrl( fullUrl ) ).toEqual( fullUrl );
+			getLocaleSlug.mockImplementationOnce( () => 'fr' );
+			expect( localizeUrl( fullUrl ) ).toEqual( fullUrl );
+		} );
+	} );
+
+	test( 'handles invalid URLs', () => {
+		[ undefined, null, [], {}, { href: 'https://test' }, 'not-a-url', () => {} ].forEach(
+			( fullUrl ) => {
+				getLocaleSlug.mockImplementationOnce( () => 'en' );
+				expect( localizeUrl( fullUrl ) ).toEqual( fullUrl );
+				getLocaleSlug(); // make sure to consume it.
+				getLocaleSlug.mockImplementationOnce( () => 'en' );
+				expect( localizeUrl( fullUrl, 'fr' ) ).toEqual( fullUrl );
+				getLocaleSlug(); // make sure to consume it.
+				getLocaleSlug.mockImplementationOnce( () => 'fr' );
+				expect( localizeUrl( fullUrl ) ).toEqual( fullUrl );
+				getLocaleSlug(); // make sure to consume it.
+			}
+		);
+	} );
+
+	test( 'handles double localizeUrl', () => {
+		getLocaleSlug.mockImplementationOnce( () => 'de' ).mockImplementationOnce( () => 'de' );
+		expect( localizeUrl( localizeUrl( 'https://automattic.com/cookies/' ) ) ).toEqual(
+			'https://automattic.com/de/cookies/'
+		);
+		getLocaleSlug();
+		getLocaleSlug(); // make sure to consume it.
+
+		getLocaleSlug.mockImplementationOnce( () => 'de' ).mockImplementationOnce( () => 'de' );
+		expect(
+			localizeUrl( localizeUrl( 'https://en.support.wordpress.com/all-about-domains/' ) )
+		).toEqual( 'https://wordpress.com/de/support/all-about-domains/' );
+		getLocaleSlug();
+		getLocaleSlug(); // make sure to consume it.
+
+		getLocaleSlug.mockImplementationOnce( () => 'de' ).mockImplementationOnce( () => 'de' );
+		expect( localizeUrl( localizeUrl( 'https://wordpress.com/' ) ) ).toEqual(
+			'https://de.wordpress.com/'
+		);
+		getLocaleSlug();
+		getLocaleSlug(); // make sure to consume it.
+
+		getLocaleSlug.mockImplementationOnce( () => 'de' ).mockImplementationOnce( () => 'de' );
+		expect( localizeUrl( localizeUrl( 'https://en.blog.wordpress.com/' ) ) ).toEqual(
+			'https://wordpress.com/blog/'
+		);
+		getLocaleSlug();
+		getLocaleSlug(); // make sure to consume it.
+	} );
+
+	test( 'trailing slash variations', () => {
+		getLocaleSlug.mockImplementationOnce( () => 'de' );
+		expect( localizeUrl( 'https://automattic.com/cookies/' ) ).toEqual(
+			'https://automattic.com/de/cookies/'
+		);
+		getLocaleSlug.mockImplementationOnce( () => 'de' );
+		expect( localizeUrl( 'https://automattic.com/cookies' ) ).toEqual(
+			'https://automattic.com/de/cookies/'
+		);
```

Comment:
It's probably an artifact of serverside version. We use a permalink structure that specifies trailing slash on urls in at least some if not all wp properties, so we probably have something that cleans these up in a filter running somewhere. I only learned about this recently so don't quote me as it might be off on some detail.